AirCasting/Graph/External/ExternalGraph.swift

```diff
@@ -0,0 +1,170 @@
+            let chartDataEntry = ChartDataEntry(x: timeInterval, y: getValue(of: measurement))
+            return chartDataEntry
+        }) ?? []
+        let allLimitLines = getLimitLines()
+        uiView.limitLines = allLimitLines
+        simplifyGraphline(entries: entries, uiView: uiView)
+        context.coordinator.totalNumberOfMeasurements = entries.count
+        context.coordinator.currentThreshold = ThresholdWitness(sensorThreshold: thresholds)
+        context.coordinator.currentMeasurementsNumber = calculateSeeingPointsNumber(entries: entries, uiView: uiView)
+        context.coordinator.entries = entries
+        context.coordinator.stream = stream
+        return uiView
+    }
+    
+    func updateUIView(_ uiView: AirCastingGraph, context: Context) {
+        // This code helps us to make graph faster and snappier
+        // The -if- statement is executed on dormant and active sessions to ensure it updates graph line well
+        // The -guard- code is executed only on active sessions and on data change events
+        // (⊙_◎)
+        let thresholdWitness = ThresholdWitness(sensorThreshold: self.thresholds)
+        let counter: Int = calculateSeeingPointsNumber(entries: context.coordinator.entries!, uiView: uiView)
+        
+        if counter != context.coordinator.currentMeasurementsNumber {
+            simplifyGraphline(entries: context.coordinator.entries!, uiView: uiView)
+            context.coordinator.currentMeasurementsNumber = counter
+        }
+        
+        guard context.coordinator.currentThreshold != thresholdWitness ||
+                context.coordinator.totalNumberOfMeasurements != stream.allMeasurements?.count ||
+                stream != context.coordinator.stream else { return }
+        
+        try? uiView.updateWithThreshold(thresholdValues: thresholds.rawThresholdsBinding.wrappedValue)
+        let allLimitLines = getLimitLines()
+        uiView.limitLines = allLimitLines
+        
+        let entries = stream.allMeasurements?.sorted(by: { $0.time < $1.time }).compactMap({ measurement -> ChartDataEntry? in
+            let timeInterval = Double(measurement.time.timeIntervalSince1970)
+            let chartDataEntry = ChartDataEntry(x: timeInterval, y: getValue(of: measurement))
+            return chartDataEntry
+        }) ?? []
+        
+        simplifyGraphline(entries: entries, uiView: uiView)
+        
+        context.coordinator.entries = entries
+        context.coordinator.currentThreshold = ThresholdWitness(sensorThreshold: thresholds)
+        context.coordinator.totalNumberOfMeasurements = entries.count
+        context.coordinator.currentMeasurementsNumber = entries.count
+        context.coordinator.stream = stream
+    }
+    
+    private func simplifyGraphline(entries: [ChartDataEntry], uiView: AirCastingGraph) {
+        
+        let counter: Int = calculateSeeingPointsNumber(entries: entries, uiView: uiView)
```

Comment:
SeenPoints? VisiblePoints? I don't know what "SeeingPoint" could be :D

Reply:
Yeah, Visible sounds way better thx!